benchmark/benchmark_config_dispatch.cpp

```diff
@@ -61,15 +66,71 @@ static void BM_kernel_launch(benchmark::State& state)
     hipStreamSynchronize(stream);
 }
 
-BENCHMARK_CAPTURE(BM_host_target_arch, default_stream, stream_kind::default_stream);
-BENCHMARK_CAPTURE(BM_host_target_arch, per_thread_stream, stream_kind::per_thread_stream);
-BENCHMARK_CAPTURE(BM_host_target_arch, explicit_stream, stream_kind::explicit_stream);
-BENCHMARK_CAPTURE(BM_host_target_arch, async_stream, stream_kind::async_stream);
-BENCHMARK(BM_kernel_launch);
+#define CREATE_BENCHMARK(ST, SK)                \
+    benchmark::RegisterBenchmark(               \
+        bench_naming::format_name(              \
+            "{lvl:na"                           \
+            ",algo:" #ST                        \
+            ",cfg:default_config}"              \
+        ).c_str(),                              \
+        &BM_host_target_arch,                   \
+        SK                                      \
+    )                                           \
+
 
 int main(int argc, char** argv)
 {
+    cli::Parser parser(argc, argv);
+    parser.set_optional<size_t>("size", "size", DEFAULT_N, "number of values");
+    parser.set_optional<int>("trials", "trials", -1, "number of iterations");
+    parser.set_optional<std::string>("name_format",
+                                    "name_format",
+                                    "human",
+                                    "either: json,human,txt");
+    parser.run_and_exit_if_error();
+
+    // Parse argv
     benchmark::Initialize(&argc, argv);
-    add_common_benchmark_info();
+    const size_t size   = parser.get<size_t>("size");
+    const int    trials = parser.get<int>("trials");
+    bench_naming::set_format(parser.get<std::string>("name_format"));
+
+    std::cout << "benchmark_config_dispatch" << std::endl;
+
+    // HIP
+    hipStream_t     stream = 0; // default
+    hipDeviceProp_t devProp;
+    int             device_id = 0;
+    HIP_CHECK(hipGetDevice(&device_id));
+    HIP_CHECK(hipGetDeviceProperties(&devProp, device_id));
+    std::cout << "[HIP] Device name: " << devProp.name << std::endl;
+
+    std::vector<benchmark::internal::Benchmark*> benchmarks{
+        CREATE_BENCHMARK(default_stream, stream_kind::default_stream),
+        CREATE_BENCHMARK(per_thread_stream, stream_kind::per_thread_stream),
+        CREATE_BENCHMARK(explicit_stream, stream_kind::explicit_stream),
+        CREATE_BENCHMARK(async_stream, stream_kind::async_stream)
+    };
+
+
+    // Use manual timing
+    for(auto& b : benchmarks)
+    {
+        b->UseManualTime();
+        b->Unit(benchmark::kMillisecond);
+    }
+
+    // Force number of iterations
+    if(trials > 0)
+    {
+        for(auto& b : benchmarks)
+        {
+            b->Iterations(trials);
+        }
+    }
+
+    // Run benchmarks
     benchmark::RunSpecifiedBenchmarks();
+    return 0;
+
 }
```

Comment:
Missing newline at EoF

```suggestion
}

```